services/QuillLMS/app/models/activity.rb

```diff
@@ -128,6 +128,7 @@ class Activity < ApplicationRecord
   OLD_ELL_INTERMEDIATE_DIAGNOSTIC_ACTIVITY_ID = 1568
   OLD_ELL_ADVANCED_DIAGNOSTIC_ACTIVITY_ID = 1590
   PRE_TEST_DIAGNOSTIC_IDS = [STARTER_DIAGNOSTIC_ACTIVITY_ID, INTERMEDIATE_DIAGNOSTIC_ACTIVITY_ID, ADVANCED_DIAGNOSTIC_ACTIVITY_ID, ELL_STARTER_DIAGNOSTIC_ACTIVITY_ID, ELL_INTERMEDIATE_DIAGNOSTIC_ACTIVITY_ID, ELL_ADVANCED_DIAGNOSTIC_ACTIVITY_ID]
+  OLD_PRE_TEST_DIAGNOSTIC_IDS = [OLD_STARTER_DIAGNOSTIC_ACTIVITY_ID, OLD_INTERMEDIATE_DIAGNOSTIC_ACTIVITY_ID, OLD_ADVANCED_DIAGNOSTIC_ACTIVITY_ID, OLD_ELL_STARTER_DIAGNOSTIC_ACTIVITY_ID, OLD_ELL_INTERMEDIATE_DIAGNOSTIC_ACTIVITY_ID, OLD_ELL_ADVANCED_DIAGNOSTIC_ACTIVITY_ID]
 
```

Comment:
I would add a constant:
`ALL_PRE_TEST_DIAGNOSTIC_IDS = PRE_TEST_DIAGNOSTIC_IDS + OLD_PRE_TEST_DIAGNOSTIC_IDS`

Then in the four queries below, you rely on one constant instead of having to remember to include the old_pre